src/package/packageVersionCreate.ts

```diff
@@ -486,7 +486,7 @@ export class PackageVersionCreate {
       {
         Package: typesArr,
       },
-      [unpackagedMetadataPath]
+      [hasUnpackagedMetadata ? unpackagedMetadataPath : null]
```

Comment:
@tshopshireSalesforce while this assignment is safer, the issue is the guard on setting var `hasUnpackagedMetadata` on line 467. Having unpackagedMetadata is independent of running tests with code coverage.

Reply:
my obseravation agrees with @tshopshireSalesforce saying.  unpackagedMetadata worked only when code coverage option was provided.  now, am not sure what is the expected behavior though.  That someone from packaging can confirm

Reply:
Here's my summary (correct me if anything here is wrong):
Tyler's original change caused the regression because it created a discrepancy between the profiles in the manifest 
 and the actual profiles, when "hasUnpackagedMetadata" is identified as false due to codecoverage not being specified. In that case, the unpackageMetadataPath is **not** excluded from the manifest creation, but the path is excluded from the profile generation.
 
This fix should resolve that. 

It's possible I'm missing something, but I would think that this logic should be agnostic to whether or not the codecoverage flag is used. I understand that the unpackaged metadata dir is designed for adding metadata needed to run Apex tests and that Apex tests don't run unless the codecoverage flag is specified, but why would we want to allow profiles from a directory identified as being used for unpackaged metadata to be included in the package profiles? It's likely relatively innocent since you cannot release a package that hasn't run codecoverage.

My suggestion would be to procced with this change and I'll raise this with unlocked packaging team to handle with a separate lower priority bug.